src/main/java/io/ebeaninternal/server/transaction/DefaultTransactionThreadLocal.java

```diff
@@ -1,19 +1,25 @@
 package io.ebeaninternal.server.transaction;
 
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.persistence.PersistenceException;
+
 import io.ebeaninternal.api.SpiTransaction;
-import io.ebeaninternal.server.transaction.TransactionMap.State;
 
 /**
  * Used to store Transactions in a ThreadLocal.
  */
 public final class DefaultTransactionThreadLocal {
 
-  private static final ThreadLocal<TransactionMap> local = new ThreadLocal<TransactionMap>() {
-    @Override
-    protected synchronized TransactionMap initialValue() {
-      return new TransactionMap();
-    }
-  };
```

Comment:
Here are several misconcepts, that prevents an application server to unload ebean/webappclassloader
1. the bracket initalization constructs an anonymous class, which holds a reference to our classloader
2. the initalValue is also a class from ebean, which holds also a reference to the classloader.

If such a reference is stuck in an other thread, the GC cannot remove the classloader on redeploy.
A java.util.HashMap can be cleared by the leakDetector, so that all references can go away.